App crashes after working in the background and plays a video that was opened from a YT link a while ago on restart

### Steps to reproduce

1. Open a YT link using the app
2. Close the app
3. Open the app again and start watching another video
4. Pause the video
5. Use other apps for a while without closing LibreTube
6. Open LibreTube again - it crashes
7. Open LibreTube yet again - it starts playing the video from a YT link you opened in the first place and shows an error
8. _(Optional, may not work out)_ Use other apps for a while without closing LibreTube
9. _(Optional, may not work out)_ Open LibreTube again - it crashes
10. _(Optional, may not work out)_ Open LibreTube yet again - it opens without playing anything and shows the same error

### Expected behavior

Using other apps and returning to LibreTube should not cause any crashes. Or at least the app should open normally after such crash, without opening a video from a link I opened a while ago.

### Actual behavior

In the morning I opened a YouTube link which opened the video in LibreTube. I closed the video and the app.
In the evening I opened LibreTube and started watching another video. I left the app opened and the video paused while I used other apps. I returned to LibreTube, but it crashed, I opened it again and it opened with the video from that link I opened in the morning and displayed an error.
Than I reproduced the steps 8-10 and LibreTube opened like this:


https://github.com/libre-tube/LibreTube/assets/63293432/ede2bbda-7ee0-45fd-aab0-798578794507

The same thing occurred a couple times in the past in the stable version.


### LibreTube version

LibreTube Debug for e36656e

### Android version

Android 13

### Other details

This issue is somewhat similar to #3870 yet different because the error and the method of reproducing are different. Here's the error:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.github.libretube.debug/com.github.libretube.ui.activities.MainActivity}: kotlin.UninitializedPropertyAccessException: lateinit property exoPlayer has not been initialized
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3645)
 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3782)
 at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
 at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
 at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loopOnce(Looper.java:201)
 at android.os.Looper.loop(Looper.java:288)
 at android.app.ActivityThread.main(ActivityThread.java:7924)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property exoPlayer has not been initialized
 at com.github.libretube.ui.fragments.PlayerFragment.onDestroy(PlayerFragment.kt:643)
 at androidx.fragment.app.Fragment.performDestroy(Fragment.java:3376)
 at androidx.fragment.app.FragmentStateManager.destroy(FragmentStateManager.java:812)
 at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:346)
 at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1943)
 at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1839)
 at androidx.fragment.app.FragmentManager.execSingleAction(FragmentManager.java:1751)
 at androidx.fragment.app.BackStackRecord.commitNow(BackStackRecord.java:317)
 at com.github.libretube.helpers.NavigationHelper.navigateVideo(NavigationHelper.kt:155)
 at com.github.libretube.helpers.NavigationHelper.navigateVideo$default(NavigationHelper.kt:51)
 at com.github.libretube.ui.activities.MainActivity.loadIntentData(MainActivity.kt:418)
 at com.github.libretube.ui.activities.MainActivity.onCreate(MainActivity.kt:180)
 at android.app.Activity.performCreate(Activity.java:8342)
 at android.app.Activity.performCreate(Activity.java:8321)
 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1417)
 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3626)
 ... 12 more
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
+++ b/app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt
@@ -639,8 +639,9 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
 
         viewModel.player = null
 
+        if (this::exoPlayer.isInitialized) exoPlayer.pause()
+
         // disable the auto PiP mode for SDK >= 32
-        exoPlayer.pause()
         if (PlayerHelper.pipEnabled) {
             PictureInPictureCompat.setPictureInPictureParams(
                 requireActivity(),

